From a4d8d811c316b24a2add8d2b19a0f5248a678270 Mon Sep 17 00:00:00 2001
:
: Upstreaming needs more verbose log message and since Intel's SDM doesn't
: properly document those MSRs we need to argue about that.
:
: https://lore.kernel.org/xen-devel/20230718132334.2087-1-simon@invisiblethingslab.com/
:
From: Simon Gaiser <simon@invisiblethingslab.com>
Date: Mon, 8 Jul 2024 13:39:12 +0200
Subject: [PATCH] x86/idle: Get PC{8..10} counters for Tiger and Alder Lake

TODO

Signed-off-by: Simon Gaiser <simon@invisiblethingslab.com>
---
 xen/arch/x86/acpi/cpu_idle.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/xen/arch/x86/acpi/cpu_idle.c b/xen/arch/x86/acpi/cpu_idle.c
index 52808f980993..183db9f62c88 100644
--- a/xen/arch/x86/acpi/cpu_idle.c
+++ b/xen/arch/x86/acpi/cpu_idle.c
@@ -155,6 +155,12 @@ static void cf_check do_get_hw_residencies(void *arg)
 
     switch ( c->x86_model )
     {
+    /* Tiger Lake */
+    case 0x8C:
+    case 0x8D:
+    /* Alder Lake */
+    case 0x97:
+    case 0x9A:
     /* 4th generation Intel Core (Haswell) */
     case 0x45:
         GET_PC8_RES(hw_res->pc8);
@@ -185,9 +191,6 @@ static void cf_check do_get_hw_residencies(void *arg)
     case 0x6C:
     case 0x7D:
     case 0x7E:
-    /* Tiger Lake */
-    case 0x8C:
-    case 0x8D:
     /* Kaby Lake */
     case 0x8E:
     case 0x9E:
-- 
2.48.1

